Configuration.props

```diff
@@ -54,7 +54,7 @@
     <HostCxxName Condition=" '$(HostCxxName)' == '' ">c++</HostCxxName>
     <MakeConcurrency Condition=" '$(MakeConcurrency)' == '' And '$(HostCpuCount)' != '' ">-j$(HostCpuCount)</MakeConcurrency>
     <ManagedRuntime Condition=" '$(ManagedRuntime)' == '' And '$(OS)' != 'Windows_NT' ">mono</ManagedRuntime>
-    <ManagedRuntimeArgs Condition=" '$(ManagedRuntimeArgs)' == '' And '$(ManagedRuntime)' == 'mono' ">--debug=casts</ManagedRuntimeArgs>
+    <ManagedRuntimeArgs Condition=" '$(ManagedRuntimeArgs)' == '' And '$(ManagedRuntime)' != 'mono' ">--debug=casts</ManagedRuntimeArgs>
```

Comment:
This is wrong; we want `==`, otherwise `$(ManagedRuntimeArgs)` will be *empty* on non-Windows platforms, which isn't what we want.

For example, this PR updates `Mono.Android.targets` so that `$(ManagedRuntimeArgs)` is used to invoke `jnienv-gen.exe`:

```xml
    <Exec
	        Command="$(ManagedRuntime) $(ManagedRuntimeArgs) ..\..\bin\Build$(Configuration)\jnienv-gen.exe -o Android.Runtime\JNIEnv.g.cs --use-java-interop"
	    />
```

Yay, this is good.

*However*, the [build log](https://jenkins.mono-project.com/job/xamarin-android-pr-pipeline-release/563/) shows that `$(ManagedRuntimeArgs)` is *empty*:

```
mono "../../bin/BuildRelease/jnienv-gen.exe" Java.Interop/JniEnvironment.g.cs obj-MonoAndroid/Release//jni.c
```

We want `==`, not `!=`.